dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResource.java

```diff
@@ -0,0 +1,320 @@
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            helper.deleteSecret(secretForm, user);
+            return Response.ok(new ResponseEntityView(OK)).build(); // 200
+        } catch (Exception e) {
+            Logger.error(this.getClass(),"Exception creating secret with message: " + e.getMessage(), e);
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    @DELETE
+    @Path("/{serviceKey}/{siteId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response deleteAllServiceIntegrationSecrets(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("serviceKey") final String serviceKey,
+            @PathParam("siteId") final String siteId
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            //this will remove a specific configuration for the key and site combination. All the secrets at once will be lost.
+            helper.deleteServiceIntegrationSecrets(serviceKey, siteId, user);
+            return Response.ok(new ResponseEntityView(OK)).build(); // 200
+        } catch (Exception e) {
+            Logger.error(this.getClass(), "Exception getting service integration and secrets with message: " + e.getMessage(), e);
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    @DELETE
+    @Path("/{serviceKey}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response deleteServiceIntegration(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("serviceKey") final String serviceKey,
+            @QueryParam("removeDescriptor") final boolean removeDescriptor
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            helper.removeServiceIntegration(serviceKey, user, removeDescriptor);
+            return Response.ok(new ResponseEntityView(OK)).build(); // 200
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)